messagingpush/src/main/java/io/customer/messagingpush/CustomerIOPushNotificationService.kt

```diff
@@ -0,0 +1,20 @@
+package io.customer.messagingpush
+
+import io.customer.messagingpush.notification.PushNotificationListener
+
+/**
+ * Helper class to hold client listeners for push notifications
+ */
+object CustomerIOPushNotificationService {
+    internal var pushNotificationListener: PushNotificationListener? = null
```

Comment:
Does the SDK need to hold a reference to the listener? 

We should avoid holding any sort of strong references like this whenever possible. I think that we can avoid this reference if we design the API of this feature differently. 

If we are able to pass in a `PushNotificationListener` instance into a SDK function to be used in the function instead of having to first *set* a listener object and *then* call a function that uses the listener. 

Or, maybe we can make `PushNotificationListener.onNotificationComposed`  a function that you can override in the `CustomerIOFirebaseMessagingService`?

Reply:
Are you suggesting to handle this in similar fashion like we did for `CustomerIOUrlHandler` in `CustomerIOConfig`? I don't think that's possible since we are in `messagingpush` module in this case and the notification call hierarchy is triggered by FCM so this might be tricky.

For the other suggestion, I think that might not be a very good idea as `CustomerIOFirebaseMessagingService` uses `CustomerIOPushNotificationHandler`, and updating `CustomerIOPushNotificationHandler` to reference back `CustomerIOFirebaseMessagingService` will create a cyclic dependency which isn't a very good idea in my opinion.

However, if we can think of doing this by adding some bridge class in between, that might be a good idea than holding listeners in an object. I'll think about handling it this way, meanwhile if you have something to suggest, please feel to share your thoughts.